lib/pure/times.nim

```diff
@@ -1116,7 +1127,7 @@ when not defined(JS):
         gmtoff {.importc: "tm_gmtoff".}: clong
   else:
     type
-      StructTM {.importc: "struct tm", final.} = object
+      StructTM {.importc: "struct tm", pure, final.} = object
```

Comment:
added it mainly for consistency. there's lots of `pure` everywhere, in fact most `importc` have them.. should they all be removed instead? does `importc` imply `pure`? the manual is somewhat unclear here - "they're no longer needed for abi compatibility" - but if I create a random Nim type without `importc` and without `pure`, it will not be abi compatible with c - it will have a type header.. for example, if I'm developing a shared memory transport, I cannot simply memcpy c types, and do the same in nim without pure, and expect it to work.